### ISSUE TITLE
[FEATURE] PandasTypeSelector should error when dtypes are different in fit and transform

### ISSUE THREAD

**Issue by MBrouns** [koaning/scikit-lego#212]
In class today we had a train and test set with different datatypes. This causes the `PandasTypeSelector` to return a different amount of columns in train and test which causes sklearn to crash with the unhelpful error `columns are different in fit and transform`. 

It would be helpful for debugging that the `PandasTypeSelector` would already error earlier and specify what the differences are.
